lib/array_equals.rb

```diff
@@ -1,5 +1,21 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  
+  if array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  elsif !(array1.length == array2.length)
+    return false
+  end
+  
+  equal_array = []
```

Comment:
It seems a little wasteful to build an array here like this.  Maybe just return false if any of them are not equal.